src/hotspot/share/runtime/mutexLocker.cpp

```diff
@@ -295,10 +296,11 @@ void mutex_init() {
   MUTEX_DEFN(CodeHeapStateAnalytics_lock     , PaddedMutex  , safepoint);
   MUTEX_DEFN(ThreadsSMRDelete_lock           , PaddedMonitor, service-2); // Holds ConcurrentHashTableResize_lock
   MUTEX_DEFN(ThreadIdTableCreate_lock        , PaddedMutex  , safepoint);
-  MUTEX_DEFN(SharedDecoder_lock              , PaddedMutex  , tty-1);
+  MUTEX_DEFN(SharedDecoder_lock              , PaddedMutex  , service-5);
```

Comment:
Why this? Do we print stacks under NMT lock protection?

Reply:
Yes I think so. `MemTracker::print_containing_region` first acquires the  NMT lock then the `SharedDecoder_lock` like this:

`MemTracker::print_containing_region` -->
`PrintRegionWalker::do_allocation_site` -->
`NativeCallStack::print_frame` --> 
`Decoder::get_source_info` --> 
`Decoder::shared_decoder_lock()`

Reply:
Thanks. But eew. Okay, ugly but not easy to fix.
Can you please add a trailing comment behind SharedDecoder lock that it needs to be lower than the NMT lock?

Reply:
yes ok, I can add an explanatory comment there